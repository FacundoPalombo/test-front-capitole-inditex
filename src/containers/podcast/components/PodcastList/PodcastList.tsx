import React from 'react';
import styles from './styles.module.scss';

import { useQuery } from '@tanstack/react-query';
import { useParams } from 'react-router-dom';

import { episodes as episodesQuery } from '@App/queries/podcasts';
import { EpisodesResults } from '@App/model/Episodes';

import PodcastEpisodesTable from '../PodcastEpisodesTable/PodcastEpisodesTable';
import Results from '../Results/Results';

const PodcastList = () => {
  const params = useParams();
  const { data: podcasts } = useQuery<EpisodesResults>({
    ...episodesQuery(params.podcastId),
    networkMode: 'offlineFirst',
  });

  return (
    podcasts && (
      <section
        data-testid="podcast-list"
        className={styles.podcast_list__content}
      >
        <Results results={podcasts.resultCount} />
        <PodcastEpisodesTable podcasts={podcasts.results} />
      </section>
    )
  );
};

export default PodcastList;
