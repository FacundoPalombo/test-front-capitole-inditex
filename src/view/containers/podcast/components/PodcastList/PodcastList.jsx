import React from 'react';
import { useRouteLoaderData } from 'react-router-dom';

import styles from './styles.module.scss';
import Results from '../Results';
import PodcastEpisodesTable from '../PodcastEpisodesTable';

const PodcastList = () => {
  const { podcasts } = useRouteLoaderData('podcasts');
  return (
    <section
      data-testid="podcast-list"
      className={styles.podcast_list__content}
    >
      <Results results={podcasts.resultCount} />
      <PodcastEpisodesTable podcasts={podcasts.results} />
    </section>
  );
};

export default PodcastList;
