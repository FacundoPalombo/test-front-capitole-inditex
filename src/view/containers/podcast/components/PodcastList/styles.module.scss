.podcast_list {
  &__content {
    width: 100%;
    height: auto;
    display: flex;
    align-items: flex-start;
    justify-content: stretch;
    flex-direction: column;
    gap: 24px;
    margin-bottom: 16px;
  }
}
