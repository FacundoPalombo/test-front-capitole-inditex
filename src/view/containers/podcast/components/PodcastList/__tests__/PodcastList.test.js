import React from 'react';
import { render } from '@testing-library/react';
import PodcastList from '../PodcastList';

jest.mock('../../Results', () => () => <div id="mockResults">Results</div>);
jest.mock('../../PodcastEpisodesTable', () => () => (
  <div id="mockPodcastEpisodesTable">PodcastEpisodesTable</div>
));

jest.mock('@tanstack/react-query', () => ({
  useQuery: jest.fn().mockReturnValue({ data: { data: 'mockData' } }),
}));

describe('PodcastList unit tests', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<PodcastList />);
    expect(asFragment()).toMatchSnapshot();
  });
});
