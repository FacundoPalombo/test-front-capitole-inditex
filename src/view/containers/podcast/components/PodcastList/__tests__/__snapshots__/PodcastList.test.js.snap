// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PodcastList unit tests should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="podcast_list__content"
    data-testid="podcast-list"
  >
    <div
      id="mockResults"
    >
      Results
    </div>
    <div
      id="mockPodcastEpisodesTable"
    >
      PodcastEpisodesTable
    </div>
  </section>
</DocumentFragment>
`;
